ludwig/utils/fs_utils.py

```diff
@@ -152,30 +179,30 @@ def safe_move_file(src, dst):
             raise
 
 
-def rename(src, tgt):
+def rename(src, tgt, storage_options: Optional[Dict[str, Any]] = None):
     protocol, _ = split_protocol(tgt)
     if protocol is not None:
-        fs = fsspec.filesystem(protocol)
+        fs = create_fs(protocol, storage_options)
         fs.mv(src, tgt, recursive=True)
     else:
         safe_move_file(src, tgt)
 
 
-def makedirs(url, exist_ok=False):
-    fs, path = get_fs_and_path(url)
-    fs.makedirs(path, exist_ok=exist_ok)
-    if not path_exists(url):
-        with fsspec.open(url, mode="wb"):
-            pass
```

Comment:
This creates a file, not a directory. That's why I deleted it

Reply:
There's actually an important reason why it was written this way (though unfortunately it was not commented, which would have made it clear).

Basically, S3-like filesystems have no concept of "directories". Directories are a higher level abstraction that some APIs provide on top of files which contain `/` characters to make them easier to organize. So when you call `makedir` on an S3-like filesystem, it can be a no-op (depending on how the client implements it). So the workaround (if downstream code assumes the existence of this "directory") is to create a file with the directory name instead.

Reply:
Got it! That makes a lot of sense - will add this back with comments for future Ludwig developers.